hoss/dimension_utilities.py

```diff
@@ -55,6 +59,42 @@ def is_index_subset(message: Message) -> bool:
     )
 
 
+def get_override_projected_dimensions(
+    varinfo: VarInfoFromDmr,
+    override_variable_name: str,
+) -> str:
+    """returns the x-y projection variable names that would
+    match the geo coordinate names
+
+    """
+    projection_variable_name = None
+    override_variable = varinfo.get_variable(override_variable_name)
+    if override_variable.is_latitude():
+        projection_variable_name = 'projected_y'
+    elif override_variable.is_longitude():
+        projection_variable_name = 'projected_x'
+    return projection_variable_name
+
+
+def get_override_dimensions(
+    varinfo: VarInfoFromDmr,
+    required_variables: Set[str],
+) -> set[str]:
+    """Determine the dimensions that need to be "pre-fetched" from OPeNDAP
+    If dimensions are not available, get variables that can be
+    used to generate dimensions
+    """
+    # check if coordinate variables are provided
+    # this should be be a configurable in hoss_config.json
+    try:
+        override_dimensions = varinfo.get_references_for_attribute(
+            required_variables, 'coordinates'
+        )
+        return override_dimensions
+    except AttributeError:
+        return set()
```

Comment:
But isnt returning an empty set same as not findng any coordinate variables?

Reply:
What I'm asking here is: should we let this go on, or should we raise an exception? Looking back at this, returning an empty set might be the right thing to do, because there are likely variables with anonymous (or size-only) dimensions that are not spatially gridded.

The problem with not raising an exception is that, if the variable _should_ have a `coordinates` metadata attribute and doesn't, then we will not perform a spatial subset on the variable, even though we probably should. (So the output will be wrong)

Reply:
I'm not sure I understand all the particulars here, but if the behavior is not as desired, but configuration will fix - I'm ok with that (i.e., don't worry about raising an exception).

Reply:
We should minimise our reliance on putting things in the configuration file. It is not a scalable solution when hundreds of collections may be configured to use HOSS.

Reply:
if the variable 'should' have a coordinates attribute and does not..?
Not sure what determines that it 'should'
if it does not have it - we will not do a spatial subset - which is how it should be.
if we don't have dimension scales or coordinate attributes - we assume they are not spatially subsettable